oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticIndex.java

```diff
@@ -90,9 +91,16 @@ protected IndexNode acquireIndexNode(String indexPath) {
 
     @Override
     protected String getFulltextRequestString(IndexPlan plan, IndexNode indexNode) {
-        return Strings.toString(new ElasticResultRowIterator(plan.getFilter(), getPlanResult(plan), plan,
+        ElasticRequestFacade elasticRequestFacade = new ElasticResultRowIterator(plan.getFilter(), getPlanResult(plan), plan,
                 acquireIndexNode(plan), FulltextIndex::shouldInclude, getEstimator(plan.getPlanName()))
-                .getElasticQuery(plan, getPlanResult(plan)));
+                .getElasticQueryFacade(plan, getPlanResult(plan));
+        if (elasticRequestFacade.getElasticRequest() instanceof QueryBuilder){
```

Comment:
I would try to avoid "instanceof" if possible. Not sure if it's possible here...

Reply:
Sorry I don't understand what is the reason you are now using ElasticRequestFacade (see below).